### DISCUSSION TITLE
Add an .espanso_ignore file to ignore some folders or files during reload

### DISCUSSION THREAD

**Discussion by Dlakoms** [espanso/espanso#1392] [Feature requests and Ideas]
Even if we could put scripts or else in other directories than the .espanso/ one. I would find it easier and maybe less resource intensive to use scripts directories without espanso trying to parse them looking for configs. Another benefit to this approach would be to have /config/ files or /match/ files being ignored depending on which .espanso_ignore file I'm using (great when sharing configs accross devices with different OS, .espanso directory paths, etc.)

Example : 
- If I create a new directory for my python scripts and environment configs (like .venv) inside the .espanso/ directory, Espanso will try to reload config while making updates / installations on the environment. 

While putting scripts, notes and folders on the .espanso directory is convenient. 
And while putting config/ and match/ files in this directory is the only way for espanso to reload its configuration (in absence of a config file dedicated to where configs could be stored elsewhere).

If we reserve the .espanso_ignore filename for this config file, we could also make it possible to disable it with just a filename edit like ._espanso_ignore making use of other .espanso_ignore variations instead (like one for MacOS on a machine, one for Win10 on another, etc.).

**Comment by fstamour** [comment]
I would like something similar too. I keep my espanso config in a git repository and everytime I do something with git (e.g. `git add`) espanso detect the file changes in the `.git` directory and reloads the file.

I don't think we would need a `.espansoignore` file right. We could start by watching only the config directories. And/or ignoring files and directories that starts  with a dot. Yet another alternative is to have the "ignore patterns" in the yaml.

Also, there's a very good crate just for that: https://crates.io/crates/ignore

**Comment by Kugeleis** [comment]
I just stumbled across this issue. I made the `.espanso` folder a git repo and espanso looks into the git folder. As this discussion is 2 years old: Is there a solution to it?

**Comment by smeech** [comment]
There is no direct solution to the Issue as I understand it.

Espanso reloads every time a `.yml` or `yaml` file in the Espanso file tree is changed, although it doesn't actually load matches outside the `espanso/match/` directory, or those in `.yaml` files, or files prefixed with `_`.

If you don't want this to happen, you can store the file outside the Espanso tree, and amend [`includes:`](https://espanso.org/docs/configuration/include-and-exclude/) or [`imports:`](https://espanso.org/docs/matches/organizing-matches/#imports) when you want to use it. There are a few requests to _include_ these in Espanso's file watcher, however.
